src/image.ts

```diff
@@ -46,11 +72,46 @@ const srcsetWithWidths = (widths: number[]) => (url: string, mappings: ImageMapp
 const srcset: (url: string, mappings: ImageMappings) => string =
     srcsetWithWidths(defaultWidths)
 
+const parseImage = (docParser: (html: string) => DocumentFragment) =>
+    (element: BlockElement): Option<Image> => {
+    const masterAsset = element.assets.find(asset => asset?.typeData?.isMaster);
+    const data = element.imageTypeData;
+    
+    return fromNullable(masterAsset).andThen(asset => {
+        if (
+            asset?.file === undefined ||
+            asset?.typeData?.width === undefined ||
+            asset?.typeData?.height === undefined
+        ) {
+            return new None();
+        }
+
+        return new Some({
+            src: asset.file,
+            alt: fromNullable(data?.alt),
+            width: asset.typeData.width,
+            height: asset.typeData.height,
+            caption: fromNullable(data?.caption).fmap(docParser),
+            credit: fromNullable(data?.credit).andThen<string>(
+                c => data?.displayCredit ? new Some(c) : new None()
+            ),
+            nativeCaption: data?.caption ?? '',
```

Comment:
Should we make this optional like caption and credit? https://github.com/guardian/bridget/blob/master/thrift/native.thrift#L14

Reply:
Good idea, done 👍